modules/qna/src/backend/transfer.ts

```diff
@@ -1,66 +1,58 @@
 import _ from 'lodash'
 
-import * as parsers from './parsers.js'
+import { QnaItem, QnaEntry } from './qna'
+import Storage from './storage'
 
-const ANSWERS_SPLIT_CHAR = '†'
-
-export const importQuestions = async (questions, params) => {
-  const { storage, config, format = 'json', statusCallback, uploadStatusId } = params
+function assertIsArrayOfQnaItems(data: QnaItem[]): data is QnaItem[] {
+  if (!_.isArray(data)) {
+    return false
+  }
+  return data.reduce((accumulator: boolean, current: QnaItem) => accumulator && assertIsQnaItem(current), true)
+}
 
-  statusCallback(uploadStatusId, 'Calculating diff with existing questions')
+function assertIsQnaItem(data: QnaItem): data is QnaItem {
+  return _.isString(data.id) && assertIsQnaEntry(data.data)
+}
 
-  const existingQuestions = (await storage.fetchAllQuestions()).map(item =>
-    JSON.stringify(_.omit(item.data, 'enabled'))
+function assertIsQnaEntry(data: QnaEntry): data is QnaEntry {
+  return (
+    data &&
+    (data.action === 'text' || data.action === 'redirect' || data.action === 'text_redirect') &&
+    data.answers !== undefined &&
+    _.isString(data.category) &&
+    data.questions !== undefined &&
+    _.isString(data.redirectFlow) &&
+    _.isString(data.redirectNode)
   )
+}
+
+export const importQuestions = async (questions: QnaItem[], storage, config, statusCallback, uploadStatusId) => {
+  statusCallback(uploadStatusId, 'Validating data format')
+
+  // Not a perfect assertion, but better than nothing
+  if (!assertIsArrayOfQnaItems(questions)) {
+    throw new Error('Unable to convert file content to QNA items')
+  }
+
+  statusCallback(uploadStatusId, 'Calculating diff with existing questions')
 
-  const hasCategory = storage.hasCategories()
-  const parsedQuestions =
-    typeof questions === 'string' ? parsers[`${format}Parse`](questions, { hasCategory }) : questions
-  const questionsToSave = parsedQuestions.filter(item => !existingQuestions.includes(JSON.stringify(item)))
+  const existingQuestionItems = (await (storage as Storage).fetchQNAs()).map(item => item.id)
+  const itemsToSave = questions.filter(item => !existingQuestionItems.includes(item.id))
+  const entriesToSave = itemsToSave.map(q => q.data)
 
   if (config.qnaMakerApiKey) {
-    return storage.insert(questionsToSave.map(question => ({ ...question, enabled: true })))
+    return (storage as Storage).insert(entriesToSave.map(question => ({ ...question, enabled: true })))
   }
```

Comment:
I think qnaMakerApi was removed elsewhere, if it's not referenced anywhere we can remove it

Reply:
yes just remove any reference to qnaMaker